### PR TITLE
resource/alicloud_slb: Deprecates it and using new resource name alicloiud_slb_load_balancer instead; Deprecates attribute name and using new load_balancer_name instead; Upgrades its dependence sdk

### PR DESCRIPTION
**Pull request by liuri1205**
…ttributes.

### CODE REVIEW

File: alicloud/data_source_alicloud_slb_load_balancers_test.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+			"ids.#":   "0",
+			"names.#": "0",
+		}
+	}
+	var alicloudSlbLoadBalancersCheckInfo = dataSourceAttr{
+		resourceId:   "data.alicloud_slb_load_balancers.default",
+		existMapFunc: existAlicloudSlbLoadBalancersDataSourceNameMapFunc,
+		fakeMapFunc:  fakeAlicloudSlbLoadBalancersDataSourceNameMapFunc,
+	}
+	alicloudSlbLoadBalancersCheckInfo.dataSourceTestCheck(t, rand, idsConf, tagsConf, nameRegexConf, statusConf, allConf)
```
Comment (lixue323) [inline alicloud/data_source_alicloud_slb_load_balancers_test.go:104]: 老代码中的测试参数，在新的当中也支持一下吧

File: website/docs/r/slb_load_balancer.html.markdown [hunk 2]
```diff
@@ -1,24 +1,27 @@
 ---
 subcategory: "Server Load Balancer (SLB)"
```
Comment (lixue323) [inline website/docs/r/slb_load_balancer.html.markdown:2]: 老文档应该是要保留，标记弃用,
-> **DEPRECATED:** This resource has been renamed to [alicloud_slb_load_balancer](https://www.terraform.io/docs/providers/alicloud/r/slb_load_balancer.html) from version 1.122.0.

File: alicloud/resource_alicloud_slb_load_balancer.go [hunk 3]
```diff
@@ -0,0 +1,715 @@
...
+					return false
+				},
+			},
+			"delete_protection": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"off", "on"}, false),
+				Default:      "off",
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if d.Get("payment_type").(string) == "Subscription" {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_slb_load_balancer.go:86]: This also need to checking instance_charge_type. And you can refer to https://github.com/aliyun/terraform-provider-alicloud/blob/master/alicloud/diff_suppress_funcs.go#L375

File: alicloud/resource_alicloud_slb_load_balancer.go [hunk 4]
```diff
@@ -0,0 +1,715 @@
...
+	request["RegionId"] = client.RegionId
+	if v, ok := d.GetOk("resource_group_id"); ok {
+		request["ResourceGroupId"] = v
+	}
+
+	if v, ok := d.GetOk("slave_zone_id"); ok {
+		request["SlaveZoneId"] = v
+	}
+
+	if v, ok := d.GetOk("specification"); ok {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_slb_load_balancer.go:278]: Please using load_balancer_spec

File: alicloud/resource_alicloud_slb_load_balancer.go [hunk 5]
```diff
@@ -0,0 +1,715 @@
...
+					if d.Get("payment_type").(string) == "Subscription" {
+						return true
+					}
+					return false
+				},
+			},
+			"internet_charge_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"PayByBandwidth", "PayByTraffic"}, false),
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_slb_load_balancer.go:95]: Please using `validation.StringInSlice([]string{"PayByBandwidth", "PayByTraffic"}, true)` attempt to correct the previous values. It aims to keep sync the value with OpenAPI parameter values.

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: alicloud/data_source_alicloud_slb_load_balancers.go [hunk 6]
```diff
@@ -0,0 +1,680 @@
...
+
+func dataSourceAlicloudSlbLoadBalancers() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudSlbLoadBalancersRead,
+		Schema: map[string]*schema.Schema{
+			"master_availability_zone": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Removed:  "Field 'master_availability_zone' has removed from v1.122.0 and replace by 'master_zone_id'.",
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_slb_load_balancers.go:25]: Please update the next version to 1.123.0

**Pull request merged**